### DISCUSSION TITLE
Eslint extension in vscode doesn't show rule errors in scripts where configuration file is in an antecedent folder while cli eslint does.

### DISCUSSION THREAD

**Discussion by IntoLighter** [eslint/eslint#16064] [Q&A] [answered]
This is the structure of my project 
<img width="290" alt="image" src="https://user-images.githubusercontent.com/57671679/176141722-2cb64c0e-8000-4850-b9af-cd6eef739618.png">
As you can see, it's a root js file and template react project.
My configuration require non-use of semicolons. React files use semicolons by default, but vscode doesn't show any errors:
<img width="391" alt="image" src="https://user-images.githubusercontent.com/57671679/176142990-1d011e2b-64b2-49cd-a946-e9f4d8e792b1.png">
But cli does:
<img width="442" alt="image" src="https://user-images.githubusercontent.com/57671679/176143046-34d2c09d-82b0-4824-ba6d-2f060fd8af65.png">
This is output log
```
[Info  - 4:34:21 PM] ESLint server is starting
[Info  - 4:34:21 PM] ESLint server running in node v16.13.0
[Info  - 4:34:21 PM] ESLint server is running.
[Info  - 4:34:22 PM] ESLint library loaded from: /Users/a/RiderProjects/Test/node_modules/eslint/lib/api.js
Warning: React version not specified in eslint-plugin-react settings. See https://github.com/jsx-eslint/eslint-plugin-react#configuration .
[Info  - 4:34:27 PM] ESLint library loaded from: /Users/a/RiderProjects/Test/React/node_modules/eslint/lib/api.js
[Info  - 4:34:28 PM] standard: 	Configuration for rule "accessor-pairs" is invalid: 	Value {"setWithoutGet":true,"enforceForClassMembers":true,"getWithoutSet":false} should NOT have additional properties.  Referenced from: /Users/a/RiderProjects/Test/.eslintrc.js
Uncaught exception received.
Error: standard:
	Configuration for rule "accessor-pairs" is invalid:
	Value {"setWithoutGet":true,"enforceForClassMembers":true,"getWithoutSet":false} should NOT have additional properties.

Referenced from: /Users/a/RiderProjects/Test/.eslintrc.js
    at validateRuleOptions (/Users/a/RiderProjects/Test/React/node_modules/eslint/lib/config/config-validator.js:133:19)
    at /Users/a/RiderProjects/Test/React/node_modules/eslint/lib/config/config-validator.js:176:9
    at Array.forEach (<anonymous>)
    at validateRules (/Users/a/RiderProjects/Test/React/node_modules/eslint/lib/config/config-validator.js:175:30)
    at Object.validate (/Users/a/RiderProjects/Test/React/node_modules/eslint/lib/config/config-validator.js:262:5)
    at loadFromDisk (/Users/a/RiderProjects/Test/React/node_modules/eslint/lib/config/config-file.js:521:19)
    at load (/Users/a/RiderProjects/Test/React/node_modules/eslint/lib/config/config-file.js:564:20)
    at /Users/a/RiderProjects/Test/React/node_modules/eslint/lib/config/config-file.js:430:36
    at Array.reduceRight (<anonymous>)
    at applyExtends (/Users/a/RiderProjects/Test/React/node_modules/eslint/lib/config/config-file.js:408:26)
```

My `.eslintrc`:
```
module.exports = {
  env: {
    browser: true
  },
  extends: ['plugin:react/recommended', 'standard'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaFeatures: {
      jsx: true
    },
    ecmaVersion: 'latest',
    sourceType: 'module'
  },
  plugins: ['react', '@typescript-eslint'],
  rules: {
    'space-before-function-paren': [
      2,
      {
        anonymous: 'never',
        named: 'never',
        asyncArrow: 'always'
      }
    ],
    quotes: [2, 'single', { allowTemplateLiterals: true }],
    'no-multiple-empty-lines': 0,
    'eol-last': 0,
    'prefer-const': 0,
    'generator-star-spacing': [2, { before: false, after: true }],
    'comma-dangle': 0
  }
}
```

**Comment by mdjermanovic** [comment, marked as answer]
> [Info  - 4:34:21 PM] ESLint server is starting
> [Info  - 4:34:21 PM] ESLint server running in node v16.13.0
> [Info  - 4:34:21 PM] ESLint server is running.
> [Info  - 4:34:22 PM] **ESLint library loaded from: /Users/a/RiderProjects/Test/node_modules/eslint/lib/api.js**
> Warning: React version not specified in eslint-plugin-react settings. See https://github.com/jsx-eslint/eslint-plugin-react#configuration .
> [Info  - 4:34:27 PM] **ESLint library loaded from: /Users/a/RiderProjects/Test/React/node_modules/eslint/lib/api.js**
> [Info  - 4:34:28 PM] standard: 	Configuration for rule "accessor-pairs" is invalid: 	Value {"setWithoutGet":true,"enforceForClassMembers":true,"getWithoutSet":false} should NOT have additional properties.  Referenced from: /Users/a/RiderProjects/Test/.eslintrc.js
> Uncaught exception received.
> Error: standard:
> 	Configuration for rule "accessor-pairs" is invalid:
> 	Value {"setWithoutGet":true,"enforceForClassMembers":true,"getWithoutSet":false} should NOT have additional properties.
>
> Referenced from: /Users/a/RiderProjects/Test/.eslintrc.js
>    at validateRuleOptions (**/Users/a/RiderProjects/Test/React/node_modules/eslint/lib/config/config-validator.js**:133:19)


Can you check which version of ESLint you have installed in `/Users/a/RiderProjects/Test/React/node_modules/eslint`? By the stack trace, this looks like an old version (< 6.0.0), and the crash you're seeing in the output log is most likely because your configuration enables some rule options that didn't exist in that version.

I'm not sure why vscode extension first loads ESLint from `/Users/a/RiderProjects/Test/node_modules/eslint/lib/api.js` and then again from `/Users/a/RiderProjects/Test/React/node_modules/eslint/lib/api.js`, but that would be a question for https://github.com/microsoft/vscode-eslint repo.

**Reply by IntoLighter** [reply]
Thanks. It was the problem that it used `eslint` from `React/node_modules` which major version was 5. It was installed by `npx create-react-app` and, unfortunately, I didn't find why it was such out-of-date.
